src/Core/Grid/Data/Factory/CustomerGroupsDataFactory.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Grid\Data\Factory;
+
+use PrestaShop\PrestaShop\Core\Grid\Data\GridData;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollection;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+
+class CustomerGroupsDataFactory implements GridDataFactoryInterface
```

Comment:
You can remove this class it does nothing more than the generic `DoctrineGridDataFactory` we only implement additional grid data factory when we need to decorate or extend the general behaviour That's not your case, and you don't even use this class in your service definitions